string preview
uint32 batt1